src/IrisConvert/IrisConvert.php

```diff
@@ -14,101 +13,103 @@ class IrisConvert
-    private $progress;
     private $array_of_sizes_to_be_converted = [];
-    private $array_of_sizes_to_be_deleted   = [];
+    private $image_to_regenerate = [];
     private static $allowed_mime_type = ['image/jpeg', 'image/jpg', 'image/png'];
 
-    public function __construct($attachment_id, $metadata, $progress)
+    /**
+     * @param $attachment_id
+     * @param $metadata
+     */
+
+    public function __construct($attachment_id, $metadata)
     {
         $this->file_path = get_attached_file($attachment_id);
         $this->file_dirname = pathinfo($this->file_path, PATHINFO_DIRNAME);
         $this->file_ext = strtolower(pathinfo($this->file_path, PATHINFO_EXTENSION));
         $this->file_name_no_ext = pathinfo($this->file_path, PATHINFO_FILENAME);
         $this->image_metadata = $metadata;
         $this->image_id = $attachment_id;
-        $this->progress = $progress;
-        add_action('cli_init', [__CLASS__, 'generateCLI']);
     }
 
-    public function check_file_exists($attachment_id)
+    /**
+     * @param $attachment_id
+     * @return bool
+     */
+
+    public function checkFileExists($attachment_id)
     {
         $file = get_attached_file($attachment_id);
 
         if (!file_exists($file)) {
-            return;
             $message = 'The uploaded file does not exist on the server. Encoding not possible.';
             Debug::debug('The uploaded file,' . $file . ' does exist on the server. Encoding not possible.', 1);
+            return;
         }
     }
 
-    public function create_array_of_sizes_to_be_converted($metadata)
-    {
-        // // check if meta data image sizes is empty, if it is then run a wp thumbnail regenerate then continue
-        if (isset($metadata['sizes']) && empty($metadata['sizes'])) {
-            $this->regenerate_to_convert_array_of_images($metadata);
-        }
+    /**
+     * @param $attachment_id
+     * @return bool
+     */
 
-        // push original file to the array
-        array_push($this->array_of_sizes_to_be_converted, $this->file_path);
+    public function checkFileType($attachment_id)
+    {
+        $file = get_attached_file($attachment_id);
 
-        // push all created sizes of the file to the array
-        if (!empty($metadata['sizes'])) {
-            foreach ($metadata['sizes'] as $value) {
-                if (in_array($value->mime_type, self::$allowed_mime_type)) {
-                    array_push($this->array_of_sizes_to_be_converted, $this->file_dirname . '/' . $value['file']);
-                }
-            }
+        if ($this->file_ext === 'webp') {
+            $message = 'This file is already webp format.';
+            Debug::debug('This file,' . $file . ' is already converted to webp.', 1);
+            return;
         }
     }
 
+    /**
+     * @param $metadata
+     * @return array
+     */
 
-
-    public function regenerate_to_convert_array_of_images($metadata)
+    public function create_array_of_sizes_to_be_converted($metadata)
```

Comment:
Just from a consistency point, could you convert this method to a camelcase'd one?